syntax = "proto3";

package antifraud.v1;

service AntifraudAPI {
  rpc AntifraudAssesment(AntifraudAssesmentRequest) returns (AntifraudAssesmentResponse);
}

message AntifraudAssesmentRequest {
  int64 value_in_cents = 1;
  int64 transaction_timestamp = 2;
  double latitude = 3;
  double longitude = 4;
}

enum AntifraudAssesmentStatus {
  ACCEPTED = 0;
  REJECTED = 1;
}

message AntifraudAssesmentResponse {
  AntifraudAssesmentStatus status = 1;
}
